backend/src/event/Event.ts

```diff
@@ -270,6 +380,34 @@ export class Event {
     return this
   }
 
+  setByBasicCtMsg(
+    userId: number,
+    contributionId: number,
+    amount?: decimal,
+    message?: string,
+  ): Event {
+    this.setByBasicCt(userId, contributionId, amount)
+    if (message) this.message = message
+
+    return this
+  }
+
+  setByBasicCtX(
+    userId: number,
+    xUserId?: number,
+    xCommunityId?: number,
+    contributionId?: number,
+    amount?: decimal,
```

Comment:
the same here:
please define all parameters as mandatory, because there will be no contribution without an amount, a contributionId and for events of the BasicCtX type the xUserId and the xCommunityId also exists and have to be set as parameters

Reply:
true, solved